crates/kubelet/src/kubelet.rs

```diff
@@ -115,27 +120,33 @@ impl<P: Provider> Kubelet<P> {
             ..Default::default()
         };
         let mut operator_runtime = OperatorRuntime::new(&self.kube_config, operator, Some(params));
-        let operator_task = operator_runtime.start().fuse().boxed();
+        let mut operator_task = tokio::spawn(async move {
+            operator_runtime.start().await;
+            Ok::<(), anyhow::Error>(())
+        });
 
         // These must all be running for graceful shutdown. An error here exits ungracefully.
         let core = Box::pin(async {
```

Comment:
You should not need `Box::pin` here.

Reply:
We can try pulling that out or making it a normal box. If we don’t box most of these top level tasks, we end up with stack overflows on some systems

Reply:
Ah, ok. Stack overflow is a valid reason to box it.